test/utils/testconfig/testconfig.go

```diff
@@ -0,0 +1,274 @@
+	mode := StringToMode(modeString)
+
+	switch mode {
+	case OrdinaryClock, BoundaryClock, DualNICBoundaryClock, Discovery:
+		logrus.Infof("%s mode detected", mode)
+		GlobalConfig.PtpModeDesired = mode
+		GlobalConfig.Status = InitStatus
+		return GlobalConfig
+	case None:
+		logrus.Infof("No test mode specified using, %s mode. Specify the env variable PTP_TEST_MODE with one of %s, %s, %s, %s", OrdinaryClock, Discovery, OrdinaryClock, BoundaryClock, DualNICBoundaryClockString)
+		GlobalConfig.PtpModeDesired = OrdinaryClock
+		GlobalConfig.Status = InitStatus
+		return GlobalConfig
+	default:
+		logrus.Infof("%s is not a supported mode, assuming %s", mode, OrdinaryClock)
+		GlobalConfig.PtpModeDesired = OrdinaryClock
+		GlobalConfig.Status = InitStatus
+		return GlobalConfig
+	}
+}
+func GetFullDiscoveredConfig(namespace string, aClient *testclient.ClientSet, forceUpdate bool) TestConfig {
+	defer logrus.Infof("Current PTP test config=%s", &GlobalConfig)
+	if GlobalConfig.Status == DiscoveryFailureStatus ||
+		GlobalConfig.Status == DiscoverySuccessStatus && !forceUpdate {
+		return GlobalConfig
+	}
+
+	discoverPTPConfiguration(namespace, aClient)
+	return GlobalConfig
+}
+
+// Returns the slave node label to be used in the test
+func discoverPTPConfiguration(namespace string, aClient *testclient.ClientSet) {
+	var slaves []ptpv1.PtpConfig
+	var masters []ptpv1.PtpConfig
+
+	configList, err := aClient.PtpConfigs(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("error getting ptpconfig list, err=%s", err)
+	}
+	for _, config := range configList.Items {
+		for _, profile := range config.Spec.Profile {
+			if IsPtpMaster(*profile.Ptp4lOpts, *profile.Phc2sysOpts) {
+				masters = append(masters, config)
+			}
+			if IsPtpSlave(*profile.Ptp4lOpts, *profile.Phc2sysOpts) {
+				slaves = append(slaves, config)
+			}
+		}
+	}
+
+	//GlobalConfig.DiscoveredMasterPtpConfig=checkPtpProfileLabels(masters)
+	discoverMode(slaves, masters, aClient)
+}
+
+func discoverMode(slaves, masters []ptpv1.PtpConfig, aClient *testclient.ClientSet) {
+
+	if len(slaves) == 0 {
+		logrus.Warnf("No Configs present, cannot discover")
+		return
+	}
+	numBc := 0
+	pickedMasterConfig := checkPtpProfileLabels(masters, aClient)
+	GlobalConfig.DiscoveredMasterPtpConfig = pickedMasterConfig
+	for _, slave := range slaves {
+		pickedSlaveConfig := checkSinglePtpProfileLabels(slave, aClient)
+		if pickedSlaveConfig.Config.Name != "" {
+
+			masterIf := len(ptpv1.GetInterfaces(pickedSlaveConfig.Config, ptpv1.Master))
+			slaveIf := len(ptpv1.GetInterfaces(pickedSlaveConfig.Config, ptpv1.Slave))
+			GlobalConfig.Status = DiscoveryFailureStatus
+			// OC
+			if masterIf == 0 && slaveIf == 1 {
+				GlobalConfig.PtpModeDiscovered = OrdinaryClock
+				GlobalConfig.Status = DiscoverySuccessStatus
+				GlobalConfig.DiscoveredSlavePtpConfig = pickedSlaveConfig
+			}
+			// BC and Dual NIC BC
+			if masterIf > 1 && slaveIf == 1 {
```

Comment:
you can have slaveIf >1 , in that case one of the slave interfaces will be in passive mode.

Reply:
fixed